src/types/ipaddr.rs

```diff
@@ -2,17 +2,23 @@
 //!
 //! This module defines types for IPv4 and IPv6 which are simply based on the u8 arrays.
 
-use core::convert::TryFrom;
+use core::convert::{From, TryFrom};
 use core::fmt;
 use core::fmt::Write;
 
 use serde::{Serialize, Serializer};
 
 use crate::errors::Error as CrateError;
 
-#[derive(Default, Clone)]
+#[derive(Default, Clone, PartialEq)]
```

Comment:
Is this because of Clippy warning? May be then we should add `PartialEq` and `Eq` both.

Reply:
This was for comparing the enum using ==. It also helped with testing.

Reply:
This helped with writing tests. Also seemed logical.

Reply:
Not sure I am getting this. Can you please elaborate? guess [u8;4] can be easily compared for equality (and may be inequality ie. >, < as well - though that is useless). So what's the rationale for removing that `derive`?

Reply:
Sorry I was not clear, I added the `PartialEq` to help with writing tests for IPv4, and I felt that felt logical to have an operation to compare two IP Addresses. I think I'll add `Eq` too, because its an equivalent relation.

Reply:
Ignore my comments - I was wondering why you *removed* `PartialEq`? I realized, you actually added that. Makes sense totally! But yes if you add `PartialEq` you will have to add `Eq` or else the `clippy` might complaint.